Purchases/Caching/RCDeviceCache.m

```diff
@@ -101,15 +101,16 @@ - (void)cacheAppUserID:(NSString *)appUserID {
     [self.userDefaults setObject:appUserID forKey:RCAppUserDefaultsKey];
 }
 
-- (void)clearCachesForAppUserID:(NSString *)appUserID {
+- (void)clearCachesForAppUserID:(NSString *)oldAppUserID andSaveNewUserID:(NSString *)newUserID {
     @synchronized (self) {
         [self.userDefaults removeObjectForKey:RCLegacyGeneratedAppUserDefaultsKey];
-        [self.userDefaults removeObjectForKey:RCAppUserDefaultsKey];
-        [self.userDefaults removeObjectForKey:[self purchaserInfoUserDefaultCacheKeyForAppUserID:appUserID]];
+        [self.userDefaults removeObjectForKey:[self purchaserInfoUserDefaultCacheKeyForAppUserID:oldAppUserID]];
         [self clearPurchaserInfoCacheTimestamp];
         [self clearOfferingsCache];
 
-        [self deleteAttributesIfSyncedForAppUserID:appUserID];
+        [self deleteAttributesIfSyncedForAppUserID:oldAppUserID];
+
+        [self.userDefaults setObject:newUserID forKey:RCAppUserDefaultsKey];
```

Comment:
There's a function `- (void)cacheAppUserID:(NSString *)appUserID {` in this class that already does this, so this can be replaced with `[self cacheAppUserID:newUserID]`

Reply:
yep, updating